plugins/woocommerce/client/legacy/js/frontend/single-product.js

```diff
@@ -326,14 +330,79 @@ jQuery( function( $ ) {
 				}
 				captionEl.children[0].textContent = item.title;
 				return true;
-			}
+			},
+			timeToIdle: 0, // Ensure the gallery controls are always visible to avoid keyboard navigation issues.
 		}, wc_single_product_params.photoswipe_options );
 
 		// Initializes and opens PhotoSwipe.
 		var photoswipe = new PhotoSwipe( pswpElement, PhotoSwipeUI_Default, items, options );
+
+		photoswipe.listen( 'afterInit', function() {
+			self.trapFocusPhotoswipe( true );
+		});
+
+		photoswipe.listen( 'close', function() {
+			self.trapFocusPhotoswipe( false );
+		});
+
+		photoswipe.listen( 'destroy', function() {
+			var focusCurrentTargetId = setTimeout( function() {
+				currentTarget.focus();
+				clearTimeout( focusCurrentTargetId );
+			}, 500 );
```

Comment:
Out of curiosity, why is the timeout required, here? :thinking:

Reply:
@Aljullu Managing focus via JS may be tricky sometimes, especially when dealing with external plugins. By moving `currentTarget.focus();` into `setTimeout` we ensure all the DOM manipulation has finished and it's safe to do it. This delay also helps screen readers to announce the focused element correctly.

Reply:
Thanks for answering, @Manussakis! I agree that managing focus can be a bit tricky. Do you have a specific example of a plugin that might be problematic in this specific case?

In a default install, this simplified code seems to work for me:

```
photoswipe.listen( 'close', function() {
	self.trapFocusPhotoswipe( false );
	currentTarget.focus();
});
```

Do you think it would work, or are there specific cases where it doesn't? If so, it would be good to document them (maybe in code comments?) so future devs know it.